features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/internal/MiscUtils.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace.opentelemetry.internal
+
+import android.os.Build
+import com.datadog.android.api.InternalLogger
+
+internal const val NEEDS_DESUGARING_ERROR_MESSAGE =
+    "Trying to use OpenTelemetry SDK support for Android 23 and below. " +
+        "In order for this to properly work you will need to enable coreDesugaring " +
+        "in your compileOptions"
+
+internal fun <T : Any?> executeSafelyOnAndroid23AndBelow(
+    action: () -> T,
```

Comment:
you may want to place `action` as last argument to be able to use syntax like `foo(args) { lambda }` for the invocation